application/client/src/app/ui/views/toolbar/chart/zoomer/cursor/state.ts

```diff
@@ -0,0 +1,274 @@
+import { InternalAPI } from '@service/ilc';
+import { ViewContainerRef } from '@angular/core';
+import { Chart, IPositionChange } from '@ui/service/toolbar/chart';
+
+enum EChangeKind {
+    move = 'move',
+    left = 'left',
+    right = 'right',
+    undefined = 'undefined',
+}
+
+const CSettings = {
+    minSize: 20,
+};
+
+export class State {
+    public width: number = -1;
+    public left: number = 0;
+
+    private _ilc!: InternalAPI;
+    private _chart!: Chart;
+    private _detectChanges!: () => void;
```

Comment:
actually if you need many stuff from parent component you can just send into state component as it is.
In state
```
    public bind(parent: IlcInterface & ChangesDetector) {
        // and you will get access to all
        parent.ilc().services.system.bridge;
        parent.env().subscriber.register(...);
        parent.detectChanges();
    }

```
In parent component:
```
    public ngAfterContentInit(): void {
        ...
        this.state.bind(this);
        ...
    }
```